src/test/scala/mesosphere/marathon/core/deployment/DeploymentPlanTest.scala

```diff
@@ -422,6 +422,36 @@ class DeploymentPlanTest extends UnitTest with GroupCreation {
       Then("The deployment is not valid")
       validate(plan2)(f.validator).isSuccess should be(false)
     }
+
+    "Deployment plan treats change with 0 instances as scale-only change" in {
```

Comment:
nit: `s/change with 0 instances/change *and* scale down to 0 instances` to make the case clear